osquery/tables/system/darwin/connected_displays.mm

```diff
@@ -0,0 +1,215 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#import <AppKit/NSDocument.h>
+#import <Foundation/Foundation.h>
+
+#include <osquery/core/tables.h>
+#include <osquery/logger/logger.h>
+
+@interface SPDocument : NSDocument {
+}
+- (id)reportForDataType:(id)arg1;
+@end
+
+namespace osquery {
+namespace tables {
+
+QueryData genConnectedDisplays(QueryContext& context) {
+  QueryData results;
+  @autoreleasepool {
+    Row r;
+
+    // BEWARE: Because of the dynamic nature of the calls in this function, we
+    // must be careful to properly clean up the memory. Any future modifications
+    // to this function should attempt to ensure there are no leaks.
+    CFURLRef bundle_url = CFURLCreateWithFileSystemPath(
+        kCFAllocatorDefault,
+        CFSTR("/System/Library/PrivateFrameworks/SPSupport.framework"),
+        kCFURLPOSIXPathStyle,
+        true);
+
+    if (bundle_url == nullptr) {
+      LOG(INFO) << "Error parsing SPSupport bundle URL";
+      return results;
+    }
+
+    CFBundleRef bundle = CFBundleCreate(kCFAllocatorDefault, bundle_url);
+    CFRelease(bundle_url);
+    if (bundle == nullptr) {
+      LOG(INFO) << "Error opening SPSupport bundle";
+      return results;
+    }
+
+    CFBundleLoadExecutable(bundle);
+
+    std::function<void()> cleanup = [&]() {
+      CFBundleUnloadExecutable(bundle);
+      CFRelease(bundle);
+    };
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Warc-performSelector-leaks"
```

Comment:
Generally not a fan of silencing warnings/error, but I guess...we are silencing these because we don't know the selector before hand? Do you know if a static selector might work?

Reply:
We are silencing because we don't know the selector beforehand. I was not able to get a static selector to work here.

Reply:
I did try to  get the static selector work, but it's a bit of a rabbit hole.

I think this is fine for now, small nit -- can you please add a comment in the code on why we silence the errors? Thanks!